File: move-zeroes/move-zeroes.cs
public class Solution {
    public void MoveZeroes(int[] nums) 
    {
        int zeroIndex = -1;
        for(int i =0; i<nums.Length;i++)
        {
            if(nums[i] == 0)
            {
                for(int j = i+1; j<nums.Length;j++)
                {
                    if(nums[j] != 0)
                    {
                        nums[i] = nums[j];
                        nums[j] = 0;
                        break;
                    }
                }
            }
        }    
    }
}

File: move-zeroes/move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let len = nums.length, nonZeroIndex = 0, zeroCount = 0;
    
    if(len === 1 || !nums.includes(0))
        return nums;
    
    
    for(let i = 0; i < len; i++)
    {
        if(nums[i] !== 0)
        {
            nums[nonZeroIndex] = nums[i];
            nonZeroIndex++;
        }
        else
        {
            zeroCount++;
        }
    }
    for(let i = 0; i < zeroCount ;i++)
    {
        nums[len-1-i] = 0;
    }
};